sql-native/src/main/java/org/ballerinalang/sql/datasource/SQLDatasource.java

```diff
@@ -135,28 +155,109 @@ public static SQLDatasource retrieveDatasource(SQLDatasource.SQLDatasourceParams
         return sqlDatasourceToBeReturned;
     }
 
+    public static Connection getConnection(TransactionResourceManager trxResourceManager, BObject client,
+            SQLDatasource datasource)
+            throws SQLException {
+        Connection conn;
+        try {
+            boolean isInTransaction = trxResourceManager.isInTransaction();
+            if (!isInTransaction) {
```

Comment:
```suggestion
            if (!trxResourceManager.isInTransaction()) {
```

Reply:
We will change

Reply:
Changed. commit id is 24c39ff738c064c229053d918be3e4d986350b6a